crypto/ake/enclave/src/lib.rs

```diff
@@ -671,77 +660,39 @@ impl<EI: EnclaveIdentity> AkeEnclaveState<EI> {
         Ok((report, report_contents))
     }
 
-    /// Verify a quote
-    pub fn verify_quote(
-        &self,
-        quote: Quote,
-        qe_report: Report,
-        report_data: &EnclaveReportDataContents,
-    ) -> Result<IasNonce> {
-        // Is the qe_report for our enclave?
-        qe_report.verify()?;
+    /// Verify an enclave's attestation evidence
+    pub fn verify_attestation_evidence(&self, attestation_evidence: DcapEvidence) -> Result<()> {
+        let DcapEvidence {
+            quote,
+            collateral,
+            report_data,
+        } = attestation_evidence.clone();
 
         let nonce = report_data.nonce();
 
         let mut quote_pending = self.quote_pending.lock()?;
 
-        let report = quote_pending.pop(nonce);
+        let report = quote_pending.pop(nonce).ok_or(Error::InvalidState)?;
 
-        // If we found a matching entry in our cache, then we upgrade it to the
-        // IasPending state object, and insert it into the IAS pending cache. If
-        // the upgrade fails, it's because the quote and qe_report don't match,
-        // in which case we should simply abort the attempt because there's
-        // nothing we can do to fix it, and leaving it around is probably
-        // wrong.
-        if let Some(report) = report {
-            let mut ias_pending = self.ias_pending.lock()?;
-            let mut csprng = McRng::default();
-            let ias_nonce = loop {
-                let ias_nonce = IasNonce::new(&mut csprng)?;
-                if !ias_pending.contains(&ias_nonce) {
-                    break ias_nonce;
-                }
-            };
-            // Ensure the quote contains our report, and is sane.
-            quote.verify_report(&qe_report, &report)?;
-            ias_pending.put(ias_nonce.clone(), quote);
-            return Ok(ias_nonce);
-        }
+        // Ensure the quote contains our report, and is sane.
+        let quote_report_body = quote.app_report_body();
+        let report_body = report.body();
 
-        Err(Error::InvalidState)
-    }
+        if &report_body != quote_report_body {
+            return Err(VerifierError::Verification("Report mismatch".to_string()).into());
+        }
```

Comment:
This was the logic in the legacy `quote.verify_report()`. It seemed specific to the handshaking going on in this repo so didn't try to bring down into the SGX repo.